### PR TITLE
Simplify variable spelling in *.ftl to access one value parameter

### LINKED ISSUE #79: Step loading variable from template enhancement
Currently if i need to load variable from template using step
https://github.com/vividus-framework/vividus/blob/master/vividus/src/main/java/org/vividus/bdd/steps/BddVariableSteps.java#L94-L95
**Step:**
```
Given I initialize the $scopes variable `request` using template `request.ftl` with parameters:
|key|
|value|
```

I have to refer parameter in template like:

**request.ftl**
```
{
    "token" : ${parameters.key[0]} 
}
```

In order to simplify API for one-row parameters we should support referring one-row parameters  using key and index
```
{
    "token" : ${key[0]} 
}
```

### PR DESCRIPTION
**Pull request by EDbarvinsky**
Enhancement #79

### CODE REVIEW

File: vividus/src/main/java/org/vividus/bdd/steps/BddVariableSteps.java [hunk 1]
```diff
@@ -96,9 +97,20 @@
...
+        Optional<String> firstKey = keySet.stream().findFirst();
+        if (keySet.size() == 1 && dataModel.get(firstKey.orElse(null)).size() == 1)
+        {
+            String key = firstKey.get();
+            value = freemarkerProcessor
+                    .process(templatePath, Map.of(key, dataModel.get(key).get(0)), StandardCharsets.UTF_8);
+        }
+        else
+        {
+            value = freemarkerProcessor.process(templatePath, Map.of("parameters", dataModel), StandardCharsets.UTF_8);
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/bdd/steps/BddVariableSteps.java]: create a single map containing both multidimensional and simple parameters and pass it to the template

File: vividus/src/main/java/org/vividus/bdd/steps/BddVariableSteps.java [hunk 2]
```diff
@@ -96,9 +96,20 @@
...
+        if (keySet.size() == 1 && dataModel.entrySet().iterator().next().getValue().size() == 1)
+        {
+            String key = keySet.stream().findFirst().get();
+            valuesMap = Map.of(parametersKey, dataModel, key, dataModel.get(key).get(0));
+        }
+        else
+        {
+            valuesMap = Map.of(parametersKey, dataModel);
+        }
+        String value = freemarkerProcessor.process(templatePath, valuesMap, StandardCharsets.UTF_8);
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/bdd/steps/BddVariableSteps.java]:
```java
Map<String, List<String>> dataModel = MapUtils.convertExamplesTableToMap(templateParameters);
String parametersKey = "parameters";
Map<String, Object> parameters = new HashMap<>();
parameters.put(parametersKey, dataModel);
if (!dataModel.contains(parametersKey))
{
    parameters.putAll(dataModel);
}
String value = freemarkerProcessor.process(templatePath, parameters, StandardCharsets.UTF_8);
```
?

Comment (EDbarvinsky) [reply vividus/src/main/java/org/vividus/bdd/steps/BddVariableSteps.java]: @valfirst  this implementation requires  refer to the variable as key[0] (using index)

Comment (valfirst) [reply vividus/src/main/java/org/vividus/bdd/steps/BddVariableSteps.java]: @EDbarvinsky that's expected

**Pull request merged**